test/common/router/router_test.cc

```diff
@@ -641,7 +641,7 @@ TEST_F(RouterTest, AddCookie) {
               Http::ConnectionPool::Callbacks& callbacks) -> Http::ConnectionPool::Cancellable* {
             response_decoder = &decoder;
             callbacks.onPoolReady(encoder, cm_.conn_pool_.host_, upstream_stream_info_);
-            return &cancellable_;
+            return nullptr;
```

Comment:
If I return the cancellable_ here, the assertion fails. I am guessing when the cancellable is returned, something is calling newStream() on it without a subsequent onPoolReady() or onPoolFailure().

@alyssawilk is this correct in the context of this test?

Reply:
Unsure of the details but I'm pretty sure the only thing that really matters in this test is the Max-Age=1337; etc. cookie logic below :-)

Reply:
Ah ok, great.